### PR TITLE
Update reference docs - remove metrics items

### PR DESCRIPTION
**Pull request by ericvn**
Please provide a description for what this PR is for.

This is a start of getting the update_ref_docs working with *master*.
Since mixer was already removed, some additional references needed to be removed from the script that was pulling in adapters from other repos.

Some work remains with removing some additional deprecated `mixer` pages, but that's outside the work in this PR.

### CODE REVIEW

**Comment by ericvn** [comment]
I did end up deleting 2 files:
- docs/tasks/policy-enforcement/denial-and-list/index.md
- docs/tasks/policy-enforcement/rate-limiting/index.md
that were marked as deprecated. I'm not sure what they are being replaced with so maybe we need a bigger TODO for that.
Note that we do point to rate limiting in a few places like `Getting Started` and I removed that as well.

I also removed some features from the feature list. Three were deprecated, but two weren't:
- Statsd Integration | Stable
- Stackdriver Integration | Beta

**Comment by ericvn** [comment]
@howardjohn @douglas-reid Would you look at this and make sure we aren't doing something excessive removing these pages from the istio.io docs. I assume the rate-limiting page would get updated with whatever comes from https://github.com/istio/istio/issues/25725.

**Comment by douglas-reid** [comment]
fwiw, I think we can remove `Statsd Integration` from our public-facing docs. I don't know of any existing tests that validate that behavior -- do you @howardjohn ?  Our `Stackdriver Integration` can probably be promoted to Stable at this point.

File: content/en/about/feature-stages/index.md [hunk 1]
```diff
@@ -55,12 +55,10 @@ Below is our list of existing features and their current phases. This informatio
 |-------------------|-------------------
 | [Prometheus Integration](/docs/tasks/observability/metrics/querying-metrics/) | Stable
 | [Local Logging (STDIO)](/docs/tasks/observability/mixer/logs/collecting-logs/) | Stable
-| [Statsd Integration](/docs/reference/config/policy-and-telemetry/adapters/statsd/) | Stable
 | [Client and Server Telemetry Reporting](/docs/reference/config/policy-and-telemetry/) | Stable
 | [Service Dashboard in Grafana](/docs/tasks/observability/metrics/using-istio-dashboard/) | Stable
 | [Istio Component Dashboard in Grafana](/docs/tasks/observability/metrics/using-istio-dashboard/) | Stable
 | [Distributed Tracing](/docs/tasks/observability/distributed-tracing/) | Stable
-| [Stackdriver Integration](/docs/reference/config/policy-and-telemetry/adapters/stackdriver/) | Beta
```
Comment (douglas-reid) [inline content/en/about/feature-stages/index.md:63]: this should be kept, but maybe file a bug to provide better docs?

File: content/en/about/feature-stages/index.md [hunk 2]
```diff
@@ -55,12 +55,10 @@ Below is our list of existing features and their current phases. This informatio
...
 | [Distributed Tracing to Zipkin / Jaeger](/docs/tasks/observability/distributed-tracing/) | Beta
 | [Logging with Fluentd](/docs/tasks/observability/mixer/logs/fluentd/) | Beta
```
Comment (douglas-reid) [inline content/en/about/feature-stages/index.md:63]: this should be removed.

File: content/en/about/feature-stages/index.md [hunk 3]
```diff
@@ -55,12 +55,10 @@ Below is our list of existing features and their current phases. This informatio
 |-------------------|-------------------
 | [Prometheus Integration](/docs/tasks/observability/metrics/querying-metrics/) | Stable
 | [Local Logging (STDIO)](/docs/tasks/observability/mixer/logs/collecting-logs/) | Stable
```
Comment (douglas-reid) [inline content/en/about/feature-stages/index.md:57]: this can be removed now

File: content/en/about/feature-stages/index.md [hunk 4]
```diff
@@ -55,12 +55,10 @@ Below is our list of existing features and their current phases. This informatio
...
-| [Statsd Integration](/docs/reference/config/policy-and-telemetry/adapters/statsd/) | Stable
 | [Client and Server Telemetry Reporting](/docs/reference/config/policy-and-telemetry/) | Stable
 | [Service Dashboard in Grafana](/docs/tasks/observability/metrics/using-istio-dashboard/) | Stable
 | [Istio Component Dashboard in Grafana](/docs/tasks/observability/metrics/using-istio-dashboard/) | Stable
```
Comment (douglas-reid) [inline content/en/about/feature-stages/index.md:60]: Maybe remove? This links to the same docs, and the dashboards have shifted a bit with the removal of all components except for Pilot.

File: scripts/grab_reference_docs.sh [hunk 5]
```diff
@@ -34,13 +34,8 @@ REPOS=(
     https://github.com/istio/istio.git@"${SOURCE_BRANCH_NAME}"
     https://github.com/istio/api.git@"${SOURCE_BRANCH_NAME}"
     https://github.com/istio/proxy.git@"${SOURCE_BRANCH_NAME}"
-    https://github.com/apigee/istio-mixer-adapter.git@master
-    https://github.com/osswangxining/alicloud-istio-grpcadapter.git@master
-    https://github.com/vmware/wavefront-adapter-for-istio.git@master
-    https://github.com/layer5io/layer5-istio-adapter.git@master
     https://github.com/apache/skywalking-data-collect-protocol.git@master
     https://github.com/ibm-cloud-security/app-identity-and-access-adapter.git@master
```
Comment (douglas-reid) [inline scripts/grab_reference_docs.sh:38]: why not remove these as well?

File: content/en/docs/reference/config/security/authorization-policy/index.html [hunk 6]
```diff
@@ -24,9 +24,19 @@
...
-to create an allow policy. The default action is &ldquo;ALLOW&rdquo; but it is useful
-to be explicit in the policy.</p>
+<p>Istio Authorization Policy also supports the AUDIT action to decide whether to log requests.
+AUDIT policies do not affect whether requests are allowed or denied to the workload.
+Requests will be allowed or denied based solely on ALLOW and DENY policies.</p>
+
+<p>A request will be internally marked that it should be audited if there is an AUDIT policy on the workload that matches the request.
+A separate plugin must be configured and enabled to actually fulfill the audit decision and complete the audit behavior.
+The request will not be audited if there are no such supporting plugins enabled.
+Currently, the only supported plugin is the <a href="/docs/reference/config/proxy_extensions/stackdriver/">Telemetry v2 Stackdriver</a> plugin.</p>
```
Comment (douglas-reid) [inline content/en/docs/reference/config/security/authorization-policy/index.html:34]: I'm not sure where this came from but I'd like to remove the `Telemetry v2` before `Stackdriver`

File: content/en/about/feature-stages/index.md (hunk 2 above)
Comment (ericvn) [reply content/en/about/feature-stages/index.md:63]: Leaving removed per second comment.

File: content/en/about/feature-stages/index.md (hunk 1 above)
Comment (ericvn) [reply content/en/about/feature-stages/index.md:63]: The referenced docs, mixer/adapter/stackdriver/config/adapter.stackdriver.config.pb.html, were moved in the big removal PR, https://github.com/istio/istio/pull/25669. https://github.com/istio/istio/issues/26993 was created for a new doc to reference.

File: content/en/about/feature-stages/index.md (hunk 2 above)
Comment (ericvn) [reply content/en/about/feature-stages/index.md:63]: removed

File: content/en/about/feature-stages/index.md (hunk 4 above)
Comment (ericvn) [reply content/en/about/feature-stages/index.md:60]: removed

File: content/en/docs/reference/config/security/authorization-policy/index.html (hunk 6 above)
Comment (ericvn) [reply content/en/docs/reference/config/security/authorization-policy/index.html:34]: This comes from istio.io/api. https://github.com/istio/api/pull/1618 created to remove.

File: content/en/about/feature-stages/index.md (hunk 1 above)
Comment (ericvn) [reply content/en/about/feature-stages/index.md:63]: Also made `Stable`

File: content/en/about/feature-stages/index.md (hunk 3 above)
Comment (ericvn) [reply content/en/about/feature-stages/index.md]: Removed in an earlier PR. Picked up removal in a rebase.

File: scripts/grab_reference_docs.sh (hunk 5 above)
Comment (ericvn) [reply scripts/grab_reference_docs.sh]: Removed

**Pull request merged**